app/.eslintrc.js

```diff
@@ -0,0 +1,320 @@
+            "error",
+            {
+                "accessibility": "explicit",
+                "overrides": {
+                    "accessors": "no-public",
+                    "constructors": "no-public"
+                }
+            }
+        ],
+        "@typescript-eslint/member-delimiter-style": [
+            "error",
+            {
+                "multiline": {
+                    "delimiter": "semi",
+                    "requireLast": true
+                },
+                "singleline": {
+                    "delimiter": "comma",
+                    "requireLast": false
+                }
+            }
+        ],
+        "@typescript-eslint/member-ordering": [
+            "error",
+            {
+                "classes": [
+                    "public-static-field",
+                    "public-instance-field",
+                    "private-static-field",
+                    "private-instance-field",
+                    "field",
+                    "constructor",
+                    "public-static-method",
+                    "public-instance-method",
+                    "private-static-method",
+                    "private-instance-method",
+                    "method"
+                ],
+                "interfaces": [
+                    "field",
+                    "constructor",
+                    "method",
+                    "signature"
+                ],
+                "typeLiterals": [
+                    "field",
+                    "constructor",
+                    "method",
+                    "signature"
+                ]
+            }
+        ],
+        "@typescript-eslint/method-signature-style": [
+            "error",
+            "method"
+        ],
+        "@typescript-eslint/naming-convention": [
+            "error",
+            {
+                "selector": "variableLike",
+                "format": ["camelCase"]
+            },
+            {
+                "selector": "memberLike",
+                "format": ["camelCase"]
+            },
+            {
+                "selector": "typeLike",
+                "format": ["PascalCase"]
+            },
+            {
+                "selector": "variable",
+                "format": ["camelCase", "UPPER_CASE"]
+            },
+            {
+                "selector": "parameter",
+                "format": ["camelCase"],
+                "leadingUnderscore": "allow"
+            },
+            {
+                "selector": "memberLike",
+                "modifiers": ["private"],
+                "format": ["camelCase"],
+                "leadingUnderscore": "allow"
+            },
+            {
+                "selector": "property",
+                "modifiers": ["static", "readonly"],
+                "types": ["boolean", "string", "number"],
+                "format": ["UPPER_CASE"]
+            },
+            {
+                "selector": "enumMember",
+                "format": ["UPPER_CASE"]
```

Comment:
On the fence about ENUMs being uppercase, this looks fucky when it comes to say the faction names.

Reply:
Enum members have conventionally always been in UPPER_CASE. Check out the Java docs for example: https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html

Reply:
Alright, I'll let this one go as PHP Constants are uppercase as well.